spec/controllers/api/items_controller_spec.rb

```diff
@@ -112,6 +112,29 @@
             items_param = serialize_to_hash(items_param)
             expect(saved_items).to match_array(items_param)
           end
+          it 'should store revisions matching changes' do
+            @controller = Api::AuthController.new
+            post :sign_in, params: test_user_credentials
+
+            @controller = Api::ItemsController.new
+            request.headers['Authorization'] = "bearer #{JSON.parse(response.body)['token']}"
+
+            # Serializing the items into an array of hashes
+            items_param = test_items.limit(1).to_a.map(&:serializable_hash)
+            items_param[0]['content'] = 'This is the new content.'
+
+            post :sync, params: { sync_token: '', cursor_token: '', limit: 5, api: '20190520', items: items_param }
+
+            items_param[0]['content'] = 'This is yet another new content.'
+            post :sync, params: { sync_token: '', cursor_token: '', limit: 5, api: '20190520', items: items_param }
+
+            item = Item.where(uuid: items_param[0]['uuid']).first
+
+            revisions = item.revisions
+            expect(revisions.count).to equal(3)
```

Comment:
This is failing for me. It is expected to be 3 but got 2 instead.

Reply:
Maybe the 2 consecutive POST requests are throttled by the sync manager? Meaning they happen too fast one after another. Maybe adding a `sleep` line between those codes make sense - @mobitar wdyt?

Reply:
Hmm as far as I know these requests should be synchronous.

Reply:
works on my local setup and CI tests are successfull as well. Are we investigating this further?

Btw. the count should be 3 (1 revision when creating, 2 after udpates)

Reply:
I think we should investigate this, I ran the tests multiple times and it seems this particular test is failing randomly. Here's the output from `rspec`:

![image](https://user-images.githubusercontent.com/5891646/83642398-c9e55880-a57c-11ea-82f2-d0282db957d0.png)

Reply:
🤔 will give this a look

Reply:
Found the reason for this.

The Sync Manager is discarding requests that are 1 second apart from what it already has. In the test I was making the same request twice. So what I did to fix this was to make the second request with the item that I retrieved as a response from the first request.

source: https://github.com/standardnotes/syncing-server/blob/develop/lib/sync_engine/2019_05_20/sync_manager.rb#L78-L90